pkg/ipam-node/cleaner/cleaner_test.go

```diff
@@ -55,58 +58,73 @@ var _ = Describe("Cleaner", func() {
+			session, err := store.Open(ctx)
 			Expect(err).NotTo(HaveOccurred())
+			// this will create empty pool config
+			session.SetLastReservedIP(testPool3, net.ParseIP("192.168.33.100"))
+
+			cleaner := cleanerPkg.New(k8sClient, store, poolManager, time.Millisecond*100, 3)
+
+			pod1UID := createPod(testPodName1, testNamespace)
+			_ = createPod(testPodName2, testNamespace)
 
 			// should keep these reservations
-			Expect(store.Reserve(testPool1, "id1", testIFName, types.ReservationMetadata{
+			Expect(session.Reserve(testPool1, "id1", testIFName, types.ReservationMetadata{
 				CreateTime:   time.Now().Format(time.RFC3339Nano),
 				PodUUID:      pod1UID,
 				PodName:      testPodName1,
 				PodNamespace: testNamespace,
 			}, net.ParseIP("192.168.1.100"))).NotTo(HaveOccurred())
 
-			Expect(store.Reserve(testPool1, "id2", testIFName, types.ReservationMetadata{},
+			Expect(session.Reserve(testPool1, "id2", testIFName, types.ReservationMetadata{},
 				net.ParseIP("192.168.1.101"))).NotTo(HaveOccurred())
 
 			// should remove these reservations
-			Expect(store.Reserve(testPool1, "id3", testIFName, types.ReservationMetadata{
+			Expect(session.Reserve(testPool1, "id3", testIFName, types.ReservationMetadata{
 				CreateTime:   time.Now().Format(time.RFC3339Nano),
 				PodName:      "unknown",
 				PodNamespace: testNamespace,
 			}, net.ParseIP("192.168.1.102"))).NotTo(HaveOccurred())
-			Expect(store.Reserve(testPool2, "id4", testIFName, types.ReservationMetadata{
+			Expect(session.Reserve(testPool2, "id4", testIFName, types.ReservationMetadata{
 				CreateTime:   time.Now().Format(time.RFC3339Nano),
 				PodName:      "unknown2",
 				PodNamespace: testNamespace,
 			}, net.ParseIP("192.168.2.100"))).NotTo(HaveOccurred())
-			Expect(store.Reserve(testPool2, "id5", testIFName, types.ReservationMetadata{
+			Expect(session.Reserve(testPool2, "id5", testIFName, types.ReservationMetadata{
 				CreateTime:   time.Now().Format(time.RFC3339Nano),
 				PodUUID:      "something", // differ from the reservation
 				PodName:      testPodName2,
 				PodNamespace: testNamespace,
 			}, net.ParseIP("192.168.2.101"))).NotTo(HaveOccurred())
 
-			Expect(store.Commit()).NotTo(HaveOccurred())
+			Expect(session.Commit()).NotTo(HaveOccurred())
 
 			go func() {
 				cleaner.Start(ctx)
 			}()
 			Eventually(func(g Gomega) {
-				store, err := storeMgr.Open(ctx)
+				store, err := store.Open(ctx)
 				g.Expect(err).NotTo(HaveOccurred())
 				defer store.Cancel()
 				g.Expect(store.GetReservationByID(testPool1, "id1", testIFName)).NotTo(BeNil())
 				g.Expect(store.GetReservationByID(testPool1, "id2", testIFName)).NotTo(BeNil())
 				g.Expect(store.GetReservationByID(testPool1, "id3", testIFName)).To(BeNil())
 				g.Expect(store.GetReservationByID(testPool2, "id4", testIFName)).To(BeNil())
 				g.Expect(store.GetReservationByID(testPool2, "id5", testIFName)).To(BeNil())
+				// pool1 has entries, should be kept in the store
```

Comment:
imo it would be clearer  if we check what pools are left in store by listing and checking the pools.
there is a bit of a "hidden" knowledge here that if last reserved IP is nil then the only reason for this is that the pool does not exist.

Reply:
Thanks, make sense. fixed